src/ledger/LedgerManagerImpl.cpp

```diff
@@ -733,6 +735,20 @@ LedgerManagerImpl::applyBufferedLedgers()
         return;
     }
 
+    auto& lcd = mSyncingLedgers.front();
+    auto& bl = mApp.getBucketManager().getBucketList();
```

Comment:
code to apply a ledger can be shared here (and included in the `postOnMainThreadWithDelay` below), that way we also only have one sequence of events happening when replaying buffered ledgers

Reply:
I created an `ApplyLedgerWork` and am calling it using `ConditionalWork` in `ApplyCheckpointWork`, which fit in well. To use a work object in `applyBufferedLedgers()`, my current solution is to pass a callback into `ApplyLedgerWork`, which will trigger after the ledger is closed so I can call `applyBufferedLedgers()` again. Is there a better way to be notified when a "work" is done here?

Reply:
It seems like the cleanest thing to do would be to have `ApplyBufferedLedgers` work which just keeps applying ledgers from the queue until it's empty (then we don't need to use the hacky `applyBufferedLedgers` call anymore). That new work can be put into a `WorkSequence` that does DownloadApplyTxsWork->ApplyBufferedLedgers (dispatched in CatchupWork). What do you think @sisuresh @MonsieurNicolas ?

Reply:
That sounds like the right way to do this. One concern I have is that `mSyncingLedgers` lives in `LedgerManagerImpl`. We would have to expose it to allow `ApplyBufferedLedgersWork` to access the buffered ledgers. Is that an issue?

Reply:
We could maybe add `LedgerManagerImpl::getNextBufferedLedger`, which `ApplyBufferedLedgersWork` would call?

Reply:
Yeah I think something like that makes sense